src/commands/generateToken.ts

```diff
@@ -30,6 +31,7 @@ export const builder: yargs.CommandBuilder<{}, GenerateTokenArguments> = (yargs)
     array: true,
     description: 'the domains that the client will be allowed to make requests from',
   });
+  yargs.option('t', { alias: 'resource-types', type: 'array', array: true, description: 'the product types the client will be allowed to access' });
```

Comment:
It should really not be resource/product type as its hard to infer the meaning (rasterized vector is a product type, so is it a valid value?).
something like realm, domain, zone might be better.